packages/near-membrane-base/src/membrane.ts

```diff
@@ -661,6 +662,17 @@ export function createMembraneMarshall() {
             return desc;
         }
 
+        function isPlainObjectOrEquivalent(targetPointer: Pointer): boolean {
+            const proto = getLocalValue(foreignCallableGetPrototypeOf(targetPointer));
+            return (
+                // POJOs or objects without __proto__
+                proto === ObjectPrototype ||
+                proto === null ||
```

Comment:
this is not sufficient to cover objects from same domain iframes, but it is a good first step IMO. again, this is the slow path, and therefore we can be more creative here.